storm-core/src/jvm/org/apache/storm/utils/NimbusClient.java

```diff
@@ -78,16 +78,13 @@ public static NimbusClient getConfiguredClientAs(Map conf, String asUser) {
         for (String host : seeds) {
             int port = Integer.parseInt(conf.get(Config.NIMBUS_THRIFT_PORT).toString());
             NimbusSummary nimbusSummary;
-            NimbusClient client = null;
-            try {
-                client = new NimbusClient(conf, host, port, null, asUser);
+            try(NimbusClient client = new NimbusClient(conf, host, port, null, asUser)) {
                 nimbusSummary = client.getClient().getLeader();
                 if (nimbusSummary != null) {
                     String leaderNimbus = nimbusSummary.get_host() + ":" + nimbusSummary.get_port();
                     LOG.info("Found leader nimbus : {}", leaderNimbus);
                     if (nimbusSummary.get_host().equals(host) && nimbusSummary.get_port() == port) {
                         NimbusClient ret = client;
-                        client = null;
                         return ret;
```

Comment:
I guess `ret` will be closed before returning: setting client to null was the trick to not close the instance.